Grumble, one more, `test_kdeoth.py::test_kde_2d_weighted[20]`, max rtol `5e-14`, actual rtol `5.41601795e-14`:

<details>

```
=================================== FAILURES ===================================
___________________________ test_kde_2d_weighted[20] ___________________________

n_basesample = 20

    @pytest.mark.parametrize("n_basesample",
                             [
                                20,
                                pytest.param(500, marks=[pytest.mark.xslow])
                             ]
    )
    def test_kde_2d_weighted(n_basesample):
        #some basic tests comparing to normal distribution
        rng = np.random.RandomState(8765678)
    
        mean = np.array([1.0, 3.0])
        covariance = np.array([[1.0, 2.0], [2.0, 6.0]])
    
        # Need transpose (shape (2, 500)) for kde
        xn = rng.multivariate_normal(mean, covariance, size=n_basesample).T
        wn = rng.rand(n_basesample)
    
        # get kde for original sample
        gkde = stats.gaussian_kde(xn, weights=wn)
    
    
        # evaluate vs multivariate normal, using the kde definition
        xx = np.asarray([[1, 2], [3, 4], [5, 6]])
        arg = xx[:, None, :] - gkde.dataset.T
        pdf = stats.multivariate_normal.pdf
>       assert_allclose(
            gkde(xx.T),
            np.sum(pdf(arg, cov=gkde.covariance) * gkde.weights, axis=-1),
            rtol=5e-14
        )

arg        = array([[[-0.63625347, -2.3968446 ],
        [-0.41148197, -0.88159254],
        [-0.446744  , -2.50212357],
        [ ...1558893],
        [ 5.77468618,  8.54482731],
        [ 3.16139021,  1.11491235],
        [ 3.75340578,  0.31340689]]])
covariance = array([[1., 2.],
       [2., 6.]])
gkde       = <scipy.stats._kde.gaussian_kde object at 0xd716af60>
mean       = array([1., 3.])
n_basesample = 20
pdf        = <bound method multivariate_normal_gen.pdf of <scipy.stats._multivariate.multivariate_normal_gen object at 0xe8de6d38>>
rng        = RandomState(MT19937) at 0xBEDC24A8
wn         = array([0.1166544 , 0.74884774, 0.91201815, 0.54322018, 0.07071762,
       0.68950958, 0.5055778 , 0.03749941, 0.945306...11, 0.31213806, 0.73633977, 0.65168469, 0.79232667,
       0.6871915 , 0.5136075 , 0.91087284, 0.12660732, 0.10222211])
xn         = array([[ 1.63625347,  1.41148197,  1.446744  ,  0.52757942,  1.20456194,
        -0.12204139,  1.71134526,  0.29448312...2024,  4.30304287, 10.00974115, -0.24975732,
         3.28452833,  2.78441107, -2.54482731,  4.88508765,  5.68659311]])
xx         = array([[1, 2],
       [3, 4],
       [5, 6]])

scipy/stats/tests/test_kdeoth.py:181: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<function assert_allclose.<locals>.compare at 0xd71538e8>, array([5.35496592e-02, 3.93564062e-04, 6.68839968e-14]), array([5.35496592e-02, 3.93564062e-04, 6.68839968e-14]))
kwds = {'err_msg': '', 'verbose': True, 'header': 'Not equal to tolerance rtol=5e-14, atol=0', 'equal_nan': True, ...}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
                   ^^^^^^^^^^^^^^^^^^^
E           AssertionError: 
E           Not equal to tolerance rtol=5e-14, atol=0
E           
E           Mismatched elements: 1 / 3 (33.3%)
E           Max absolute difference among violations: 3.62244928e-27
E           Max relative difference among violations: 5.41601795e-14
E            ACTUAL: array([5.354966e-02, 3.935641e-04, 6.688400e-14])
E            DESIRED: array([5.354966e-02, 3.935641e-04, 6.688400e-14])

args       = (<function assert_allclose.<locals>.compare at 0xd71538e8>, array([5.35496592e-02, 3.93564062e-04, 6.68839968e-14]), array([5.35496592e-02, 3.93564062e-04, 6.68839968e-14]))
func       = <function assert_array_compare at 0xeb57e2a8>
kwds       = {'err_msg': '', 'verbose': True, 'header': 'Not equal to tolerance rtol=5e-14, atol=0', 'equal_nan': True, ...}
self       = <contextlib._GeneratorContextManager object at 0xeb56ba80>

/opt/python/cp312-cp312/lib/python3.12/contextlib.py:81: AssertionError
=========================== short test summary info ============================
FAILED scipy/stats/tests/test_kdeoth.py::test_kde_2d_weighted[20] - Assertion...
```

</details>


That didn't show up in the last two runs, but I guess we're bumping that one as well now.

--- a/doc/source/dev/toolchain.rst
+++ b/doc/source/dev/toolchain.rst
@@ -129,7 +129,7 @@ As explained in more detail below, the current minimal compiler versions are:
 ==========  ===========================  ===============================  ============================
  Compiler    Default Platform (tested)    Secondary Platform (untested)    Minimal Version
 ==========  ===========================  ===============================  ============================
- GCC         Linux                        AIX, Alpine Linux, OSX           GCC 9.x
+ GCC         Linux                        AIX, Alpine Linux, OSX           GCC 10.3
  LLVM        OSX                          Linux, FreeBSD, Windows          LLVM 12.x
  MSVC        Windows                      -                                Visual Studio 2019 (vc142)
 ==========  ===========================  ===============================  ============================
@@ -147,7 +147,6 @@ Currently, SciPy wheels are being built as follows:
 =========================   ==============================   ====================================   =============================
  Platform                    `CI`_ `Base`_ `Images`_          Compilers                              Comment
 =========================   ==============================   ====================================   =============================
- Linux x86                   ``ubuntu-22.04``                 GCC 10.2.1                             ``cibuildwheel``
  Linux arm                   ``docker-builder-arm64``         GCC 11.3.0                             ``cibuildwheel``
  OSX x86_64 (OpenBLAS)       ``macos-15-intel``               Apple clang 13.1.6                     ``cibuildwheel``
  OSX x86_64 (Accelerate)     ``macos-15-intel``               Apple clang 15.0.0                     ``cibuildwheel``
@@ -371,7 +370,7 @@ AIX, Alpine Linux and FreeBSD.
 .. _13.x release: https://www.freebsd.org/releases/13.2R/relnotes/
 .. _freebsd-port: https://ports.freebsd.org/cgi/ports.cgi?query=gcc
 
-All the currently lowest-supported compiler versions (GCC 9, LLVM 14,
+All the currently lowest-supported compiler versions (GCC 10, LLVM 14,
 VS2019 with vc142) have full support for the C++17 *core language*,
 which can therefore be used unconditionally.
 However, as of mid-2024, support for the entirety of the C++17 standard library

--- a/scipy/optimize/tests/test_minpack.py
+++ b/scipy/optimize/tests/test_minpack.py
@@ -941,7 +941,7 @@ class TestCurveFit:
                         jac=jac2, absolute_sigma=absolute_sigma)
 
                 assert_allclose(popt1, popt2, rtol=1.4e-7, atol=1e-14)
-                assert_allclose(pcov1, pcov2, rtol=1.4e-7, atol=1e-14)
+                assert_allclose(pcov1, pcov2, rtol=1.5e-7, atol=1e-14)
 
     @pytest.mark.parametrize("absolute_sigma", [False, True])
     def test_curvefit_scalar_sigma(self, absolute_sigma):

--- a/scipy/sparse/linalg/_eigen/arpack/tests/test_arpack.py
+++ b/scipy/sparse/linalg/_eigen/arpack/tests/test_arpack.py
@@ -23,6 +23,10 @@ from scipy._lib._gcutils import assert_deallocated
 _ndigits = {'f': 3, 'd': 11, 'F': 3, 'D': 11}
 
 
+def _is_32bit():
+    return np.intp(0).itemsize < 8
+
+
 def _get_test_tolerance(type_char, mattype=None, D_type=None, which=None):
     """
     Return tolerance values suitable for a given test:
@@ -77,6 +81,11 @@ def _get_test_tolerance(type_char, mattype=None, D_type=None, which=None):
             # missing more cases, from PR 14798
             rtol *= 10
             atol *= 10
+
+    if _is_32bit():
+        # Small tolerance bumps needed on i686 linux after moving
+        # from GCC 10.2 to 14.2
+        rtol *= 2
 
     return tol, rtol, atol
 

--- a/scipy/sparse/linalg/_isolve/tests/test_iterative.py
+++ b/scipy/sparse/linalg/_isolve/tests/test_iterative.py
@@ -37,6 +37,10 @@ _SOLVERS = [bicg, bicgstab, cg, cgs, gcrotmk, gmres, lgmres,
 CB_TYPE_FILTER = ".*called without specifying `callback_type`.*"
 
 
+def _is_32bit():
+    return np.intp(0).itemsize < 8
+
+
 def _assert_success(*, A, x, b, xp, rtol=1.0, atol=0.0, less_equal=False):
     Ax = xp.squeeze(A @ x[..., xp.newaxis], axis=-1)
     residual = Ax - b
@@ -319,6 +323,9 @@ def test_maxiter(case, xp, batch_A, batch_b):
 def test_convergence(case, xp, batch_A, batch_b):
     if (case.solver is tfqmr) and ("poisson2d-F" in case.name):
         pytest.skip("Struggles to converge with single precision on some platforms")
+    if (case.solver is tfqmr) and ("rand-sym-pd-F" in case.name) and _is_32bit():
+        pytest.skip("Fails to converge on i686 (32-bit) linux")
+
     case = xp_case(case, xp, batch_A, batch_b, rng=38)
     A = case.A
 
@@ -350,13 +357,15 @@ def test_precond_dummy(case, xp, batch_A, batch_b):
         pytest.skip("Struggles to converge with single precision")
     if (case.solver is tfqmr) and ("poisson2d-F" in case.name):
         pytest.skip("Hits divide-by-zero with single precision")
+    if (case.solver is tfqmr) and ("rand-sym-pd-F" in case.name) and _is_32bit():
+        pytest.skip("Fails to converge on i686 (32-bit) linux")
     if not case.convergence:
         pytest.skip("Solver - Breakdown case, see gh-8829")
 
     rtol = 1e-8 if np.finfo(dtype).eps < 1e-8 else 1.4e-3
 
     A = case.A
-    
+
     # NOTE: the following was previously uncommented as dead code --
     # was the intention to set `A = dia_array(...)`?
 
@@ -395,7 +404,7 @@ def test_precond_inverse(case, xp, batch_A, batch_b):
         pytest.skip("specific to poisson1d and poisson2d cases")
     if case.solver is qmr:
         pytest.skip("skipped for qmr")
-    
+
     case = xp_case(case, xp, batch_A, batch_b, rng=38)
     rtol = 1e-8
 
@@ -463,7 +472,7 @@ def test_atol(solver, xp, batch_A, batch_b):
     b = 1e3 * rng.uniform(size=(*batch_b, 10))
 
     dtype = xpx.default_dtype(xp)
-    A = xp.asarray(A, dtype=dtype) 
+    A = xp.asarray(A, dtype=dtype)
     b = xp.asarray(b, dtype=dtype)
 
     tols = np.r_[0, np.logspace(-9, 2, 7), np.inf]
@@ -497,7 +506,7 @@ def test_atol(solver, xp, batch_A, batch_b):
 @pytest.mark.parametrize("batch_b", [()])
 def test_zero_rhs(solver, xp, batch_A, batch_b):
     rng = np.random.default_rng(1684414984100503)
-    dtype = xpx.default_dtype(xp) 
+    dtype = xpx.default_dtype(xp)
     A = xp.asarray(rng.random(size=(*batch_A, 10, 10)), dtype=dtype)
     A = A @ A.mT + 10 * xp.eye(10)
 

--- a/scipy/special/tests/test_data.py
+++ b/scipy/special/tests/test_data.py
@@ -551,7 +551,7 @@ BOOST_TESTS = [
         data(elliprg, 'ellint_rg_xyy_ipp-ellint_rg_xyy', (0, 1, 2), 3,
              rtol=7.5e-16),
         data(elliprg, 'ellint_rg_xy0_ipp-ellint_rg_xy0', (0, 1, 2), 3,
-             rtol=5e-16),
+             rtol=5.2e-16),
         data(elliprg, 'ellint_rg_00x_ipp-ellint_rg_00x', (0, 1, 2), 3,
              rtol=5e-16),
         data(elliprj, 'ellint_rj_data_ipp-ellint_rj_data', (0, 1, 2, 3), 4,

--- a/scipy/stats/tests/test_kdeoth.py
+++ b/scipy/stats/tests/test_kdeoth.py
@@ -22,7 +22,7 @@ def test_kde_1d():
     xx = np.asarray([0.1, 0.5, 0.9])
     loc, scale = gkde.dataset, np.sqrt(gkde.covariance)
     assert_allclose(
-        gkde(xx), 
+        gkde(xx),
         stats.norm.pdf(xx[:, None], loc=loc, scale=scale).sum(axis=-1) / gkde.n,
         rtol=5e-14
     )
@@ -65,7 +65,7 @@ def test_kde_1d_weighted():
 
     pdf = stats.norm.pdf
     assert_allclose(
-        gkde(xx), 
+        gkde(xx),
         np.sum(pdf(xx[:, None], loc=loc, scale=scale) * gkde.weights, axis=-1),
         rtol=5e-14
     )
@@ -181,7 +181,7 @@ def test_kde_2d_weighted(n_basesample):
     assert_allclose(
         gkde(xx.T),
         np.sum(pdf(arg, cov=gkde.covariance) * gkde.weights, axis=-1),
-        rtol=5e-14
+        rtol=6e-14
     )
 
     # ... and cdf

